### DISCUSSION TITLE
Source code installation execution command:：flask db upgrade

### DISCUSSION THREAD

**Discussion by yourengod** [langgenius/dify#7041] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

Source code installation execution command:：flask db upgrade

  File "/home/xx/anaconda3/envs/mydify/lib/python3.12/site-packages/sqlalchemy/engine/create.py", line 643, in connect
    return dialect.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/xx/anaconda3/envs/mydify/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 620, in connect
    return self.loaded_dbapi.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/xx/anaconda3/envs/mydify/lib/python3.12/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.OperationalError: connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

postgres cannot start

**Comment by yourengod** [comment]
sudo docker run --name postgres -e POSTGRES_PASSWORD=difyai123456 -p 5432:5432 -d postgres:15-alpine
new  question:
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) connection to server at "localhost" (127.0.0.1), port 5432 failed: FATAL:  database "dify" does not exist

**Reply by shanlusun** [reply]
The same issue, does anyone fix this ? Sames the DB table is not created and Redis is also not started.

**Comment by elise-p** [comment]
same issue right here, I tried several alternative commands etc but it doesn't work. I'm running it in windows.
